lib/Sema/CSApply.cpp

```diff
@@ -7481,6 +7481,7 @@ namespace {
         } else {
           // For other closures, type-check the body once we've finished with
           // the expression.
+          cs.setExprTypes(closure);
```

Comment:
Under what circumstances do we need to do this? We should be type checking these bodies independently. They can refer to captured data, but that data should already have a type, right?